fritzconnection/lib/fritztam.py

```diff
@@ -0,0 +1,148 @@
+"""
+Module to interact with the telephone answering machine (TAM).
+"""
+# This module is part of the FritzConnection package.
+# https://github.com/kbr/fritzconnection
+
+# License: MIT (https://opensource.org/licenses/MIT)
+# Author: Mark Ullmann
+
+
+from ..core.exceptions import FritzServiceError
+from .fritzbase import AbstractLibraryBase
+from xml.etree import ElementTree as etree
+
+
+# important: don't set an extension number here:
+SERVICE = 'X_AVM-DE_TAM'
+
+
+class FritzTAM(AbstractLibraryBase):
+    """
+    Class to interact with the integrated telephone answering machine (TAM).  All
+    parameters are optional.  If given, they have the following meaning: `fc`
+    is an instance of FritzConnection, `address` the ip of the Fritz!Box,
+    `port` the port to connect to, `user` the username, `password` the
+    password, `timeout` a timeout as floating point number in seconds,
+    `use_tls` a boolean indicating to use TLS (default False). It is
+    recommended to use a dedicated user for security reasons.  Setup of the
+    TAM is not supported, this needs to be done in the web interface.
+    """
+    # This class is adapted from the class FritzWLAN.
+    def __init__(self, *args, service=1, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.service = service
+        self._sid_token = None
+
+    def _action(self, actionname, **kwargs):
+        service = f'{SERVICE}{self.service}'
+        return self.fc.call_action(service, actionname, **kwargs)
+
+    @property
+    def _sid(self):
+        """
+        Authentication token for sid, to access certain urls. This is cached
+        internally.
+        """
+        if self._sid_token is None:
+            sidResponse = self.fc.call_action("DeviceConfig",
```

Comment:
Please use pep8 as coding style, i.e. `sid_response`. There are more issues like this in the code.